### DISCUSSION TITLE
2 queries are faster than 1 with nested read

### DISCUSSION THREAD

**Discussion by baptisteArno** [prisma/prisma#16686] [Q&A]
This:

```ts
const typebots = await prisma.typebot.findMany({
    where: {
      workspaceId: 'workspaceId',
    },
    select: { id: true },
  })

const chatsCount = await prisma.result.count({
    where: {
      typebotId: { in: typebots.map((typebot) => typebot.id) },
      hasStarted: true,
      createdAt: { gte: firstDayOfMonth, lte: lastDayOfMonth },
    },
 })
```

is significantly faster than:

```ts
const chatsCount = await prisma.result.count({
    where: {
      typebot: { workspaceId: 'workspaceId' },
      hasStarted: true,
      createdAt: { gte: firstDayOfMonth, lte: lastDayOfMonth },
    },
  })
```

Is it expected?

Dropped from ~4,000ms to ~(25 + 25) ms

**Comment by baptisteArno** [comment]
Even faster:

```ts
const chatsCount = await prisma.$transaction(async (tx) => {
    const typebotIds = await tx.typebot.findMany({
      where: {
        workspaceId: 'workspaceId',
      },
      select: { id: true },
    })

    return tx.result.count({
      where: {
        typebotId: { in: typebotIds.map((typebot) => typebot.id) },
        hasStarted: true,
        createdAt: { gte: firstDayOfMonth, lte: lastDayOfMonth },
      },
    })
  })
```

**Comment by nurul3101** [comment]
Hello @baptisteArno 👋 

The query response time is affected depending upon your where clauses, if the field with which you are filtering is a primary key or if it has an index, then it would be significantly faster than queries which do not have an index.
I would recommend you to enable [logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) to get the raw queries which are generated and then [explain analyze](https://www.postgresql.org/docs/current/sql-explain.html) them to get insights on query execution plan, this would provide insights based on your use case and database configuration.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
